### ISSUE TITLE
Review and correct Trilinos packages that #define KOKKOS_FOO macros

### ISSUE THREAD

**Issue by PhilMiller** [kokkos/kokkos#4965]
```
~/repos/Trilinos $ git grep 'define KOKKOS_' origin/develop | egrep -v 'packages/kokkos(-kernels)?/'

origin/develop:packages/minitensor/src/MiniTensor_Definitions.h:#define KOKKOS_INLINE_FUNCTION  inline
origin/develop:packages/phalanx/src/Phalanx_KokkosViewFactoryFunctor.hpp:#define KOKKOS_VIEW_FACTORY_FUNCTOR_HPP
origin/develop:packages/sacado/src/KokkosExp_View_Fad.hpp:#define KOKKOS_EXPERIMENTAL_VIEW_SACADO_FAD_HPP
origin/develop:packages/sacado/src/KokkosExp_View_Fad.hpp:#define KOKKOS_VIEW_DATA_ANALYSIS_SACADO_FAD( NS ) \
origin/develop:packages/sacado/src/KokkosExp_View_Fad_Contiguous.hpp:#define KOKKOS_EXPERIMENTAL_VIEW_SACADO_FAD_CONTIGUOUS_HPP
origin/develop:packages/sacado/src/Kokkos_DynRankView_Fad.hpp:#define KOKKOS_DYN_RANK_VIEW_SACADO_FAD_HPP
origin/develop:packages/sacado/src/Kokkos_DynRankView_Fad_Contiguous.hpp:#define KOKKOS_DYN_RANK_VIEW_SACADO_FAD_CONTIGUOUS_HPP
origin/develop:packages/sacado/src/Kokkos_LayoutContiguous.hpp:#define KOKKOS_LAYOUT_CONTIGUOUS_HPP
origin/develop:packages/sacado/src/Kokkos_LayoutNatural.hpp:#define KOKKOS_LAYOUT_NATURAL_HPP
origin/develop:packages/sacado/src/Kokkos_ViewFactory.hpp:#define KOKKOS_VIEW_FACTORY_HPP
origin/develop:packages/sacado/src/Kokkos_View_Fad.hpp:#define KOKKOS_VIEW_FAD_HPP
origin/develop:packages/sacado/src/Kokkos_View_Fad_Fwd.hpp:#define KOKKOS_VIEW_FAD_FWD_HPP
origin/develop:packages/sacado/src/Sacado_DisableKokkosCuda.hpp:// Redefine KOKKOS_*_FUNCTION macros to not include __device__
origin/develop:packages/sacado/test/performance/fenl_assembly/BoxElemFixture.hpp:#define KOKKOS_EXAMPLE_BOXELEMFIXTURE_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly/BoxElemPart.hpp:#define KOKKOS_BOXELEMPART_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly/HexElement.hpp:#define KOKKOS_HEXELEMENT_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly/fenl_functors.hpp:#define KOKKOS_EXAMPLE_FENLFUNCTORS_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly_view/BoxElemFixture.hpp:#define KOKKOS_EXAMPLE_BOXELEMFIXTURE_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly_view/BoxElemPart.hpp:#define KOKKOS_BOXELEMPART_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly_view/HexElement.hpp:#define KOKKOS_HEXELEMENT_HPP
origin/develop:packages/sacado/test/performance/fenl_assembly_view/fenl_functors.hpp:#define KOKKOS_EXAMPLE_FENLFUNCTORS_HPP
origin/develop:packages/shylu/shylu_node/basker/src/shylubasker_types.hpp:#define KOKKOS_NOINIT             Kokkos::ViewAllocateWithoutInitializing
origin/develop:packages/shylu/shylu_node/tacho/src/impl/Tacho_Util.hpp:#define KOKKOS_IMPL_PAUSE asm volatile( "pause\n":::"memory" );
origin/develop:packages/shylu/shylu_node/tacho/src/impl/Tacho_Util.hpp:#define KOKKOS_IMPL_PAUSE __asm__ __volatile__( "pause\n":::"memory" );
origin/develop:packages/shylu/shylu_node/tacho/src/impl/Tacho_Util.hpp:#define KOKKOS_IMPL_PAUSE  asm volatile( "or 27, 27, 27" ::: "memory" );
origin/develop:packages/shylu/shylu_node/tacho/src/impl/Tacho_Util.hpp:#define KOKKOS_IMPL_PAUSE
origin/develop:packages/stk/stk_simd/stk_simd_view/simd_view.hpp:#define KOKKOS_INVALID_INDEX 0
origin/develop:packages/stokhos/src/kokkos/Cuda/Stokhos_Cuda_SymmetricDiagonalSpec.hpp:#define KOKKOS_CUDA_SYMMETRIC_DIAGONAL_SPEC_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/Kokkos_AnalyzeStokhosShape.hpp:#define KOKKOS_ANALYZE_STOKHOS_SHAPE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/Kokkos_View_Utils.hpp:#define KOKKOS_VIEW_UTILS_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/Kokkos_View_Utils_Def.hpp:#define KOKKOS_VIEW_UTILS_DEF_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/KokkosExp_View_UQ_PCE_Contiguous.hpp:#define KOKKOS_EXPERIMENTAL_VIEW_UQ_PCE_CONTIGUOUS_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/Kokkos_Atomic_UQ_PCE.hpp:#define KOKKOS_ATOMIC_UQ_ATOMIC_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/Kokkos_View_UQ_PCE.hpp:#define KOKKOS_VIEW_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/Kokkos_View_UQ_PCE_Fwd.hpp:#define KOKKOS_VIEW_UQ_PCE_FWD_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/Kokkos_View_UQ_PCE_Utils.hpp:#define KOKKOS_VIEW_UQ_PCE_UTILS_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/algorithms/Kokkos_Random_UQ_PCE.hpp:#define KOKKOS_RANDOM_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/linalg/Kokkos_ArithTraits_UQ_PCE.hpp:#define KOKKOS_ARITHTRAITS_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/linalg/Kokkos_Blas1_UQ_PCE.hpp:#define KOKKOS_BLAS1_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/linalg/Kokkos_CrsMatrix_UQ_PCE.hpp:#define KOKKOS_CRSMATRIX_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/linalg/Kokkos_CrsMatrix_UQ_PCE_Cuda.hpp:#define KOKKOS_CRSMATRIX_UQ_PCE_CUDA_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/linalg/Kokkos_InnerProductSpaceTraits_UQ_PCE.hpp:#define KOKKOS_INNER_PRODUCT_SPACE_TRAITS_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/linalg/Kokkos_MV_UQ_PCE.hpp:#define KOKKOS_MV_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/pce/mpicomm/Kokkos_TeuchosCommAdapters_UQ_PCE.hpp:#define KOKKOS_TEUCHOS_COMM_ADAPTERS_UQ_PCE_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/KokkosExp_View_MP_Vector_Contiguous.hpp:#define KOKKOS_EXPERIMENTAL_VIEW_MP_VECTOR_CONTIGUOUS_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/Kokkos_Atomic_MP_Vector.hpp:#define KOKKOS_ATOMIC_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/Kokkos_Parallel_MP_Vector.hpp:#define KOKKOS_PARALLEL_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector.hpp:#define KOKKOS_VIEW_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Fwd.hpp:#define KOKKOS_VIEW_MP_VECTOR_FWD_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Interlaced.hpp:#define KOKKOS_VIEW_MP_VECTOR_INTERLACED_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/Kokkos_View_MP_Vector_Utils.hpp:#define KOKKOS_VIEW_MP_VECTOR_UTILS_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/algorithms/Kokkos_Random_MP_Vector.hpp:#define KOKKOS_RANDOM_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/linalg/Kokkos_ArithTraits_MP_Vector.hpp:#define KOKKOS_ARITHTRAITS_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/linalg/Kokkos_Blas1_MP_Vector.hpp:#define KOKKOS_BLAS1_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/linalg/Kokkos_CrsMatrix_MP_Vector.hpp:#define KOKKOS_CRSMATRIX_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/linalg/Kokkos_CrsMatrix_MP_Vector_Cuda.hpp:#define KOKKOS_CRSMATRIX_MP_VECTOR_CUDA_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/linalg/Kokkos_InnerProductSpaceTraits_MP_Vector.hpp:#define KOKKOS_INNER_PRODUCT_SPACE_TRAITS_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/linalg/Kokkos_MV_MP_Vector.hpp:#define KOKKOS_MV_MP_VECTOR_HPP
origin/develop:packages/stokhos/src/sacado/kokkos/vector/mpicomm/Kokkos_TeuchosCommAdapters_MP_Vector.hpp:#define KOKKOS_TEUCHOS_COMM_ADAPTERS_MP_VECTOR_HPP
origin/develop:packages/stokhos/test/Performance/FadMPAssembly/BoxElemFixture.hpp:#define KOKKOS_EXAMPLE_BOXELEMFIXTURE_HPP
origin/develop:packages/stokhos/test/Performance/FadMPAssembly/BoxElemPart.hpp:#define KOKKOS_BOXELEMPART_HPP
origin/develop:packages/stokhos/test/Performance/FadMPAssembly/HexElement.hpp:#define KOKKOS_HEXELEMENT_HPP
origin/develop:packages/stokhos/test/Performance/FadMPAssembly/VectorImport.hpp:#define KOKKOS_VECTORIMPORT_HPP
origin/develop:packages/stokhos/test/Performance/FadMPAssembly/fenl_functors.hpp:#define KOKKOS_EXAMPLE_FENLFUNCTORS_HPP
origin/develop:packages/stokhos/test/Performance/MPAssembly/BoxElemFixture.hpp:#define KOKKOS_EXAMPLE_BOXELEMFIXTURE_HPP
origin/develop:packages/stokhos/test/Performance/MPAssembly/BoxElemPart.hpp:#define KOKKOS_BOXELEMPART_HPP
origin/develop:packages/stokhos/test/Performance/MPAssembly/HexElement.hpp:#define KOKKOS_HEXELEMENT_HPP
origin/develop:packages/stokhos/test/Performance/MPAssembly/VectorImport.hpp:#define KOKKOS_VECTORIMPORT_HPP
origin/develop:packages/stokhos/test/Performance/MPAssembly/fenl_functors.hpp:#define KOKKOS_EXAMPLE_FENLFUNCTORS_HPP
origin/develop:packages/teuchos/kokkoscomm/src/Kokkos_MpiTypeTraits.hpp:#define KOKKOS_MPITYPETRAITS_HPP
origin/develop:packages/teuchos/kokkoscomm/src/Kokkos_TeuchosCommAdapters.hpp:#define KOKKOS_TEUCHOS_COMM_ADAPTERS_HPP
origin/develop:packages/tpetra/classic/LinAlg/Kokkos_LinAlgVersion.hpp:#define KOKKOS_LINALG_VERSION_HPP_
origin/develop:packages/tpetra/classic/LinAlg/Kokkos_MultiVectorKernelOps.hpp:#define KOKKOS_MULTIVECTOR_KERNELOPS_HPP
origin/develop:packages/tpetra/classic/NodeAPI/Kokkos_BufferMacros.hpp:#define KOKKOS_BUFFER_MACROS
origin/develop:packages/tpetra/classic/NodeAPI/Kokkos_DefaultNode.hpp:#define KOKKOS_DEFAULTNODE_HPP
origin/develop:packages/tpetra/classic/NodeAPI/Kokkos_NodeAPIConfigDefs.hpp:#define KOKKOS_NODEAPICONFIGDEFS_HPP
origin/develop:packages/tpetra/classic/src/Kokkos_ConfigDefs.hpp:#define KOKKOS_CONFIGDEFS_HPP
origin/develop:packages/tpetra/classic/src/Kokkos_Version.hpp:#define KOKKOS_VERSION_HPP_
origin/develop:packages/trilinoscouplings/examples/fenl/BelosSolve.hpp:#define KOKKOS_EXAMPLE_BELOS_SOLVE
origin/develop:packages/trilinoscouplings/examples/fenl/BoxElemFixture.hpp:#define KOKKOS_EXAMPLE_BOXELEMFIXTURE_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/BoxElemPart.hpp:#define KOKKOS_BOXELEMPART_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/CGSolve.hpp:#define KOKKOS_EXAMPLE_CG_SOLVE
origin/develop:packages/trilinoscouplings/examples/fenl/HexElement.hpp:#define KOKKOS_HEXELEMENT_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/fenl.hpp:#define KOKKOS_EXAMPLE_FENL_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/fenl_ensemble.hpp:#define KOKKOS_EXAMPLE_FENL_ENSEMBLE_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/fenl_functors.hpp:#define KOKKOS_EXAMPLE_FENLFUNCTORS_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/fenl_functors_pce.hpp:#define KOKKOS_EXAMPLE_FENLFUNCTORS_PCE_HPP
origin/develop:packages/trilinoscouplings/examples/fenl/fenl_impl.hpp:#define KOKKOS_EXAMPLE_FENL_IMPL_HPP
```

**Comment by ldh4** [comment]
> origin/develop:packages/stk/stk_simd/stk_simd_view/simd_view.hpp:#define KOKKOS_INVALID_INDEX 0

This define will be removed from stk when stk makes its next snapshot update into Trilinos.

**Comment by ajpowelsnl** [comment]
Action: contact Trilinos product owners, and ask them to help with renaming Kokkos macros.

**Comment by ajpowelsnl** [comment]
Hi @ndellingwood and @ldh4 -- do we have any progress on this issue?  If not, can you summarize what still needs to be done and guesstimate a timeline for completion?

**Comment by ldh4** [comment]
Hi @ajpowelsnl, there are no more define macro cases like `#define KOKKOS_FOO` in stk.

**Comment by ajpowelsnl** [comment]
> Hi @ajpowelsnl, there are no more define macro cases like `#define KOKKOS_FOO` in stk.

Thanks for the update, @ldh4!  It looks like there are potentially a lot more Trilinos packages with the `#define KOKKOS_FOO` macro -- is there a concrete plan and timeline for completing the rest?

**Comment by ndellingwood** [comment]
@ajpowelsnl I think the easy renaming has been completed. A sticking point was for cases where files in packages were prefixed with `Kokkos_*`, in which case there corresponding header guards use a `KOKKOS_*` convention

**Comment by ajpowelsnl** [comment]
> @ajpowelsnl I think the easy renaming has been completed. A sticking point was for cases where filenames in packages were prefixed with `Kokkos_*`, in which case their corresponding header guards use a `KOKKOS_*` convention

Thanks for the update.  Would you like to discuss the sticking points tomorrow 😁
